### PR TITLE
test/librbd/fsx: musl libc doesn't implement random_r. Use c++11 std::mt19937 generator instead.

### PR DESCRIPTION
**Pull request by dx9**
Is it possible to use random instead of random_r? random_r is a glibc extension, musl libc doesn't implement it.

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
@liewegas this is another one of the Alpine PRs. I'm not sure if this code requires the (multi-threading safe) random_r or not...

**Comment by idryomov** [comment]
random_r was added on purpose, to make sure fsx failures can actually be reproduced, see abdb168ccebadfdb59f29995cee8e6a72d164bd2.  IIRC the issue was librados stealing random numbers from fsx, in particular when running with ms_inject_socket_failures enabled.  Since of the two it is fsx that cares about reproducible sequences, I went for a private context within fsx.

**Comment by dx9** [comment]
@idryomov What do you think about using a boost RNG?

**Comment by idryomov** [comment]
@dx9 Back then fsx was a C file, but now it's actually a .cc compiled with a C++ compiler, so if it helps your project, I'd fine with boost (c++11 random would be even better).  As long you carefully test your change - we only look at fsx when something else had already failed...

**Comment by dx9** [comment]
@idryomov cool, I'll look at updating the pr using c++11 random. Thanks for the feedback. I may have some more questions about how to test this properly.

File: src/test/librbd/fsx.cc [hunk 1]
```diff
@@ -2205,14 +2191,7 @@ main(int argc, char **argv)
...
 
-	if (initstate_r(seed, rnd_state, RND_STATE_LEN, &rnd_data) < 0) {
-		prterr("initstate_r");
-		exit(1);
-	}
-	if (setstate_r(rnd_state, &rnd_data) < 0) {
-		prterr("setstate_r");
-		exit(1);
-	}
+    random_generator.seed(seed);
```
Comment (idryomov) [inline src/test/librbd/fsx.cc]: Indentation?

File: src/test/librbd/fsx.cc [hunk 2]
```diff
@@ -1416,9 +1402,9 @@ do_clone()
 	++num_clones;
 
 	if (randomize_striping) {
-		order = 18 + get_random() % 8;
-		stripe_unit = 1ull << (order - 1 - (get_random() % 8));
```
Comment (idryomov) [inline src/test/librbd/fsx.cc]: You can it much less invasive if you keep get_random() around.

Comment (dx9) [inline src/test/librbd/fsx.cc]: Thanks for taking a look. Yeah tossing get_random() wasn't a good idea. I'll put it back.

**Comment by liewegas** [comment]
@idryomov ping

**Pull request merged**